public/app/plugins/panel/geomap/utils/utils.ts

```diff
@@ -117,6 +117,25 @@ export const getNextLayerName = (panel: GeomapPanel) => {
   return `Layer ${Date.now()}`;
 };
 
+export function isSegmentVisible(
```

Comment:
Great job on this refactor! I think just adding a few unit tests for this util function would be good 👍

Reply:
I created two follow-up issues to address this. Unfortunately, because testing needs to occur postrender with canvas, Jest has limitations.

https://github.com/grafana/grafana/issues/64182
https://github.com/grafana/grafana/issues/64183